### DISCUSSION TITLE
Why is an error reported when an iOS app developed with Unity 4.x SDK is uploaded to the App Store?

### DISCUSSION THREAD

**Discussion by icywind** [AgoraIO-Extensions/Agora-Unity-Quickstart#161] [Q&A] [answered]
**Similar question: "When distributing our build for iOS, we get a x86_64 error"**

![unsupported architecture x86_64](https://web-cdn.agora.io/docs-files/1667533013233)

**Comment by icywind** [comment, marked as answer]
## Problem causes

The iOS dynamic library inside the Unity 4.x SDK contains a simulator architecture.  But the App Store does not allow the SDK of the app to have both device and a simulator architecture. Therefore, you will need to delete the binaries of the simulator.

## Solution 

 1. In XCode select **Add New Run Script Phase**  ![enter image description here](https://web-cdn.agora.io/docs-files/1667533033937)
 2. Add the following script to automatically delete the simulator architecture in the SDK when packaging.
 ```bash
 #!/bin/sh

# Strip invalid architectures

strip_invalid_archs() {
binary="$1"
echo "current binary ${binary}"
# Get architectures for current file
archs="$(lipo -info "$binary" | rev | cut -d ':' -f1 | rev)"
stripped=""
for arch in $archs; do
if ! [[ "${ARCHS}" == *"$arch"* ]]; then
if [ -f "$binary" ]; then
# Strip non-valid architectures in-place
lipo -remove "$arch" -output "$binary" "$binary" || exit 1
stripped="$stripped $arch"
fi
fi
done
if [[ "$stripped" ]]; then
echo "Stripped $binary of architectures:$stripped"
fi
}

APP_PATH="${TARGET_BUILD_DIR}/${WRAPPER_NAME}"

# This script loops through the frameworks embedded in the application and
# removes unused architectures.
find "$APP_PATH" -name '*.framework' -type d | while read -r FRAMEWORK
do
FRAMEWORK_EXECUTABLE_NAME=$(defaults read "$FRAMEWORK/Info.plist" CFBundleExecutable)
FRAMEWORK_EXECUTABLE_PATH="$FRAMEWORK/$FRAMEWORK_EXECUTABLE_NAME"
echo "Executable is $FRAMEWORK_EXECUTABLE_PATH"

strip_invalid_archs "$FRAMEWORK_EXECUTABLE_PATH"
done

```

**Reply by jtrue** [reply]
You are wonderful! Thank you for posting this

**Comment by icywind** [comment]
[中文解答](https://docs.agora.io/cn/video-call-4.x/faq/arch_unityl)

**Comment by tomkail** [comment]
Is there any chance you could add an editor script that adds this to the XCode project at build time from Unity? It's incredibly annoying!
It means that you're relying on devs to either have already encountered this problem, or to find their way to this page.
Even if you do have the solution to have to re-do it each time we replace our build folder, which is super annoying.
